services/api/service.go

```diff
@@ -1733,6 +1739,45 @@ func (api *RelayAPI) checkSubmissionSlotDetails(w http.ResponseWriter, log *logr
 		return false
 	}
 
+	// Do a check in memchache for the slot being with a registered provider. If miss, go to redis. if miss again go to postgres.
+
+	// Find the duty for the submission slot
+	api.proposerDutiesLock.RLock()
+	duty := api.proposerDutiesMap[submission.BidTrace.Slot]
+	api.proposerDutiesLock.RUnlock()
+	if duty == nil {
+		log.Info("no duty found for submission slot")
+		api.RespondError(w, http.StatusBadRequest, "no duty found for submission slot")
+		return false
+	}
+
+	if api.opts.MevCommitFiltering {
```

Comment:
To avoid nested ifs, would be cleaner to put this new logic at the end of the function, and use the following code right before the new logic

```golang
if !api.opts.MevCommitFiltering {
     	return true
}
```

Reply:
There's some code that needs to run after this return though.

Reply:
Can the `// Timestamp check` run before the new logic?